OData/src/System.Web.OData/OData/Routing/Conventions/DynamicPropertyRoutingConvention.cs

```diff
@@ -99,6 +100,9 @@ public class DynamicPropertyRoutingConvention : NavigationSourceRoutingConventio
 
                 controllerContext.RouteData.Values[ODataRouteConstants.DynamicProperty] =
                     dynamicPropertSegment.PropertyName;
+                controllerContext.RouteData.Values[ODataParameterValue.ParameterValuePrefix + ODataRouteConstants.DynamicProperty] =
+                        new ODataParameterValue(dynamicPropertSegment.PropertyName,
+                            EdmLibHelpers.GetEdmPrimitiveTypeReferenceOrNull(typeof(string)));
```

Comment:
Can you add one simple E2E test case  to verify the change works on [FromODataUri]?